test-data/unit/check-dataclasses.test

```diff
@@ -520,6 +520,26 @@ app.database_name  # E: "SpecializedApplication" has no attribute "database_name
 
 [builtins fixtures/list.pyi]
 
+[case testDataclassesInitVarsAndDefer]
+# flags: --new-semantic-analyzer
```

Comment:
Is this actually needed? Or it doesn't work well with the old analyzer?

Reply:
No, but I added it since the test case doesn't test anything useful on the old analyzer and I didn't want to add it to `check-newsemanal.test`. We may want to move test cases out of `check-newsemanal.test` to other files once we've removed the old semantic analyzer, and this is anticipating that move.